Added Scroll bar for bani reader

amitojsingh opened the pull request:
- Added scroll bar fix #303 

Previously, Android and iOS relied on their native scrollbars. However, WebView doesn’t allow reliable customization of scrollbar colors, which affected readability. As Scrollbar is hidden in Dark mode. 
To address this and improve the reading experience of the Bani, I implemented a custom scrollbar.

Before 
<img width="585" height="1266" alt="IMG_1004" src="https://github.com/user-attachments/assets/d4558bd1-58d6-4ee8-b45f-498afa95986c" />

After Adding Scrollbar
<img width="439" height="717" alt="Screenshot 2025-12-19 at 4 26 17 PM" src="https://github.com/user-attachments/assets/a2b22da4-2029-4fcd-b208-836ec96251db" />

amitojsingh committed:
Enhance ReaderScreen with custom scrollbar and WebView improvements

- Added properties to WebView to hide vertical and horizontal scroll indicators for a cleaner interface.
- Implemented a minimal custom scrollbar in gutkaScript.js for improved user experience during scrolling, ensuring visibility on both Android and iOS.
- Updated gutkahtml.js to include styles for the new scrollbar, enhancing the overall aesthetic and functionality of the ReaderScreen.

Gauravjeetsingh commented on src/ReaderScreen/utils/gutkaScript.js:
```diff
@@ -21,9 +22,21 @@ const clearScrollTimeout=()=> {
+  const p=Math.max(0,Math.min(1,(window.pageYOffset||0)/h));
+  const th=Math.max(30,(window.innerHeight/h)*window.innerHeight);
+  const t=sb.children[0];
+  if(t){t.style.height=th+"px";t.style.top=(p*(window.innerHeight-th))+"px";}
```
This feels very congested and difficult to read. Can you please fix the indentation and use proper variable names, instead of single letter variables.

amitojsingh replied:
Fixed

Gauravjeetsingh commented on src/ReaderScreen/utils/gutkaScript.js:
```diff
@@ -101,13 +114,26 @@ window.addEventListener(
-  scrollToPosition(); 
+  scrollToPosition();
+  
+  // Minimal scrollbar - works on Android and iOS
```
Can you please add some explanation, on the purpose of this code, is this removing an existing div with id `#sb` and then adding another div with same id `sb`?

amitojsingh replied:
This is added as defensive logic to handle cases where the screen may load multiple times, preventing duplicate scrollbars from being created.

Gauravjeetsingh replied:
We can't add a condition that if this div exists, do not add another? Why do we need to remove and add?

amitojsingh replied:
fixed

Gauravjeetsingh commented on src/ReaderScreen/utils/gutkahtml.js:
```diff
@@ -80,6 +80,11 @@ const htmlTemplate = (backColor, fontFace, content, theme, savePosition) => `<!D
       text-align:right
     }
+    /* Minimal scrollbar */
+    #sb { position:fixed; right:0; top:0; width:6px; height:100vh; z-index:99999; pointer-events:none; }
```
Can you fix the indentation here as well.

amitojsingh replied:
Fixed

Gauravjeetsingh reviewed (changes requested):
Mainly needs update with code indentation and variable names. 

Also, I am curious why are we creating a custom scrollbar, can you please let me know the issue with the default scrollbar?

amitojsingh committed:
Refactor scrollbar styles and update functionality in ReaderScreen

- Improved the CSS formatting for the custom scrollbar in gutkahtml.js for better readability and maintainability.
- Enhanced the updateSB function in gutkaScript.js to improve the logic for displaying the scrollbar, ensuring it adapts correctly to the content height and window size.
- Cleaned up code formatting for consistency across both files.

amitojsingh committed:
Refactor scrollbar functionality in gutkaScript.js

- Renamed variables for clarity, changing 'sb' to 'scrollbar' to enhance code readability.
- Updated the updateSB function to updateScrollbar, improving the logic for scrollbar visibility and positioning.
- Enhanced the scrollFunc to utilize the new updateScrollbar function, ensuring consistent behavior during scroll events.
- Cleaned up code formatting for better maintainability.

amitojsingh commented:
Updated the description.

Gauravjeetsingh commented on src/ReaderScreen/utils/gutkahtml.js:
```diff
@@ -80,6 +80,24 @@ const htmlTemplate = (backColor, fontFace, content, theme, savePosition) => `<!D
+    }
+    #sb-t {
+      position: absolute;
+      right: 1;
```
This should either be 0, or 1px, right? Is there a reason for it being 1 without the unit?

amitojsingh replied:
fixed.

Gauravjeetsingh commented on src/ReaderScreen/utils/gutkaScript.js:
```diff
@@ -101,20 +120,33 @@ window.addEventListener(
+  scrollbar.appendChild(scrollbarThumb);
+  document.body.appendChild(scrollbar);
+  setTimeout(updateScrollbar, 100);
+  window.addEventListener("resize", updateScrollbar);
```
Everytime the `window.onload` fires, it adds a new resize listener without removing the previous one. Maybe store its reference somewhere, and use that to add a condition that if the listener already exists, remove it before adding a new one.

amitojsingh replied:
fixed

amitojsingh committed:
Fix scrollbar styling and enhance resize handling in ReaderScreen

- Updated the scrollbar CSS in gutkahtml.js to specify the unit for the right position, ensuring proper rendering.
- Added a resize listener in gutkaScript.js to improve scrollbar responsiveness, removing any existing listener before adding a new one for better performance.
- Cleaned up code formatting for consistency and maintainability across both files.

amitojsingh committed:
Merge branch 'dev' into 303-bug-scroll-bar-is-missing

amitojsingh committed:
update comment
